sale_commission/models/account_invoice.py

```diff
@@ -126,7 +126,7 @@ def _compute_amount(self):
                     line.commission):
                 if line.commission.amount_base_type == 'net_amount':
                     subtotal = (line.object_id.price_subtotal -
-                                (line.product.standard_price *
+                                (line.object_id.price_unit *
```

Comment:
I'm checking this right now and the idea behind this is to get the net amount, so it's correct to put the cost price, not the price unit of the line.

Reply:
@pedrobaeza thanks for checking, then why are the tests failing?
As said in the description, `settlement.total + extra_total` is negative, then the `journal` is discarded.
Now the code raises an exception for that, then in the test I can use `assertRaises`, but then we will need a test for the complete flow

Reply:
@pedrobaeza I implemented your suggestion in https://github.com/OCA/commission/pull/192/commits/0fca4c3fad2d98aa96fb887c75cb46de9c9a7091.

I also moved the commission computation to `sale.commission.line.mixin` because I noticed that it was almost duplicated in `account.invoice` and `sale.order` (almost because `line.object_id` were two different models), now what is happening in the module looks a bit more clear to me; what do you think?
Thanks